### PR TITLE
Generate a node id when cloning a gui node

### LINKED ISSUE #7209: Cloned node IDs are empty hashes
**Describe the bug (REQUIRED)**
Nodes cloned with `gui.clone()` or `gui.clone_tree()` return empty hashes when getting their ID with `gui.get_id()`.

**To Reproduce (REQUIRED)**
```
local nodes = gui.clone_tree(node_to_clone)
for node_name, node in pairs(nodes) do
local node_id = gui.get_id(node)
print("node_id", node_id) -- DEBUG:SCRIPT: node_id	hash: []
end
```

**Expected behavior (REQUIRED)**
Each cloned node should be assigned a unique ID returned when using gui.get_id().

**Defold version (REQUIRED):**
1.4.0

**Platforms (REQUIRED):**
macOS, iOS, Android.

**Workaround (OPTIONAL):**
A workaround is to use gui.set_id() on each individual node. This can be time consuming if using gui.clone_tree().

**Additional context (OPTIONAL):**
https://forum.defold.com/t/getting-node-id-to-update-gooey/65744
https://github.com/britzl/defold-input/issues/5

### PR DESCRIPTION
**Pull request by britzl**
Cloned gui nodes are now automatically assigned a node id instead of an empty hash.

Fixes #7209

## PR checklist

* [x] Code
	* [x] Add engine and/or editor unit tests.
	* [x] New and changed code follows the overall code style of existing code
	* [x] Add comments where needed
* [x] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [x] Prepare pull request and affected issue for automatic release notes generator
	* [ ] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [ ] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [ ] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [ ] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

----------

Example of a well written PR description:

1. Start with the user facing changes. This will end up in the release notes.
1. Add one of the GitHub approved closing keywords
1. Optionally also add the technical changes made. This is information that might help the reviewer. It will not show up in the release notes. Technical changes are identified by a line starting with one of these:
   1. `### Technical changes` 
   1. `Technical changes:`
   2. `Technical notes:`

```
There was a anomaly in the carbon chroniton propeller, introduced in version 8.10.2. This fix will make sure to reset the phaser collector on application startup.

Fixes #1234

### Technical changes
* Pay special attention to line 23 of phaser_collector.clj as it contains some interesting optimizations
* The propeller code was not taking into account a negative phase.
```

### CODE REVIEW

File: engine/gui/src/gui.cpp [hunk 1]
```diff
@@ -4229,10 +4231,15 @@ namespace dmGui
         InternalNode* out_n = &scene->m_Nodes[index];
         memset(out_n, 0, sizeof(InternalNode));
 
+        // generate a name for the cloned node
+        char name[16];
+        sprintf(name, "node%d", g_ClonedNodeCount++);
```
Comment (britzl) [inline engine/gui/src/gui.cpp:4236]: One question is if we perhaps should chose node ids that are more uncommon than this?

* clonednode%d
* _node%d
* _clonednode%d

Comment (JCash) [reply engine/gui/src/gui.cpp:4236]: That was my first reaction, that this is easy to accidentally reproduce.
I agree, this needs to be more robust.

Comment (JCash) [reply engine/gui/src/gui.cpp:4236]: Also, use dmSnPrintf() which is cross platform

**Review by JCash — CHANGES_REQUESTED** [review]

File: engine/gui/src/gui.cpp (hunk 1 above)
Comment (britzl) [reply engine/gui/src/gui.cpp]: > I agree, this needs to be more robust.

Does `__node%d` is robust enough? Or perhaps better to add the number first? Something like `_%d_node`

**Review by JCash — APPROVED** [review]

**Pull request merged**